modin/pandas/test/test_io.py

```diff
@@ -1102,1229 +1054,1120 @@ def test_read_csv_internal(
-    make_parquet_file(NROWS)
+    def test_read_csv_categories(self, request):
+        if request.config.getoption("--simulate-cloud").lower() != "off":
+            pytest.skip(
+                "The reason of tests fail in `cloud` mode is unknown for now - issue #2340"
+            )
+        eval_io(
+            fn_name="read_csv",
+            # read_csv kwargs
+            filepath_or_buffer="modin/pandas/test/data/test_categories.csv",
+            names=["one", "two"],
+            dtype={"one": "int64", "two": "category"},
+        )
 
-    pandas_df = pandas.read_parquet(TEST_PARQUET_FILENAME, columns=["col1"])
-    modin_df = pd.read_parquet(TEST_PARQUET_FILENAME, columns=["col1"])
-    df_equals(modin_df, pandas_df)
+    @pytest.mark.parametrize("encoding", [None, "utf-8"])
+    @pytest.mark.parametrize("parse_dates", [False, ["timestamp"]])
+    @pytest.mark.parametrize("index_col", [None, 0, 2])
+    @pytest.mark.parametrize("header", ["infer", 0])
+    @pytest.mark.parametrize(
+        "names",
+        [
+            None,
+            ["timestamp", "symbol", "high", "low", "open", "close", "spread", "volume"],
+        ],
+    )
+    def test_read_csv_parse_dates(
+        self, request, names, header, index_col, parse_dates, encoding
+    ):
+        if (
+            parse_dates
+            and request.config.getoption("--simulate-cloud").lower() != "off"
+        ):
+            pytest.skip(
+                "The reason of tests fail in `cloud` mode is unknown for now - issue #2340"
+            )
 
+        if names is not None and header == "infer":
+            pytest.xfail(
+                "read_csv with Ray engine works incorrectly with date data and names parameter provided - issue #2509"
+            )
 
-def test_from_parquet_partition(make_parquet_file):
-    make_parquet_file(NROWS, directory=True)
+        eval_io(
+            fn_name="read_csv",
+            # read_csv kwargs
+            filepath_or_buffer="modin/pandas/test/data/test_time_parsing.csv",
+            names=names,
+            header=header,
+            index_col=index_col,
+            parse_dates=parse_dates,
+            encoding=encoding,
+        )
 
-    pandas_df = pandas.read_parquet(TEST_PARQUET_FILENAME)
-    modin_df = pd.read_parquet(TEST_PARQUET_FILENAME)
-    df_equals(modin_df, pandas_df)
+    @pytest.mark.skipif(Engine.get() == "Python", reason="Using pandas implementation")
+    def test_read_csv_s3(self):
+        dataset_url = "s3://noaa-ghcn-pds/csv/1788.csv"
+        pandas_df = pandas.read_csv(dataset_url)
 
+        # This first load is to trigger all the import deprecation warnings
+        modin_df = pd.read_csv(dataset_url)
 
-def test_from_parquet_partition_with_columns(make_parquet_file):
-    make_parquet_file(NROWS, directory=True)
+        # This will warn if it defaults to pandas behavior, but it shouldn't
+        with pytest.warns(None) as record:
+            modin_df = pd.read_csv(dataset_url)
```

Comment:
Why do we do this check in this case, but not in the rest.

Do we really need it?

Reply:
I think we can use regular `eval_io` function here.